src/Core/src/Dispatching/Dispatcher.Android.cs

```diff
@@ -75,18 +84,59 @@ public void Stop()
 
 			IsRunning = false;
 
-			_handler.RemoveCallbacks(OnTimerTick);
+			_handler.RemoveCallbacks(_runnable);
+
+			SetHasCallbacks(false);
 		}
 
 		void OnTimerTick()
 		{
 			if (!IsRunning)
 				return;
 
+			SetHasCallbacks(false);
+
 			Tick?.Invoke(this, EventArgs.Empty);
 
 			if (IsRepeating)
-				_handler.PostDelayed(OnTimerTick, (long)Interval.TotalMilliseconds);
+			{
+				Post();
+			}
+		}
+
+		void Post()
+		{
+			if (IsCallbackPosted())
+			{
+				return;
+			}
+
+			_handler.PostDelayed(_runnable, (long)Interval.TotalMilliseconds);
+
+			SetHasCallbacks(true);
+		}
+
+		bool IsCallbackPosted()
+		{
+			if (OperatingSystem.IsAndroidVersionAtLeast(29))
+			{
+				return _handler.HasCallbacks(_runnable);
+			}
+
+			// Below API 29 we'll manually track whether there's a posted callback
+			return _hasCallbacks;
+		}
+
+		void SetHasCallbacks(bool hasCallbacks)
+		{
+			if (OperatingSystem.IsAndroidVersionAtLeast(29))
+			{
+				return;
+			}
+
+			// We only need to worry about tracking this if we're below API 29; after that,
+			// we can just ask the Handler with the HasCallBacks() method. 
+			_hasCallbacks = hasCallbacks;
```

Comment:
Do we need to worry about thread safety here?

Reply:
I don't think so. Do you have a scenario where you think it would be a problem?

Reply:
There may be the case where you use the timer from 2 threads that could cause issues. But I am not sure if the `hasCallbacks` method is itself thread safe...

Reply:
I agree with all the above. We don't need to do locks ourselves.